test/unit_tests/braket/default_simulator/openqasm/test_interpreter.py

```diff
@@ -1997,3 +1997,124 @@ def test_basis_rotation_hermitian():
         Hadamard([2]),
         *hermitian.diagonalizing_gates(),
     ]
+
+
+@pytest.mark.parametrize(
+    "qasm, expected",
+    [
+        (
+            "\n".join(
+                [
+                    "bit[1] b;",
+                    "qubit[2] q;",
+                    "h q[0];",
+                    "h q[1];",
+                    "b[0] = measure q[0];",
+                ]
+            ),
+            [0],
+        ),
+        (
+            "\n".join(
+                [
+                    "bit[3] b;",
+                    "qubit[3] q;",
+                    "b = measure q;",
+                ]
+            ),
+            [0, 1, 2],
+        ),
+        (
+            "\n".join(
+                [
+                    "bit[1] b;",
+                    "qubit[2] q;",
+                    "h q[0];",
+                    "h q[1];",
+                    "b[0] = measure q[0:1];",
+                ]
+            ),
+            [0, 1],
+        ),
+        (
+            "\n".join(
+                [
+                    "bit[3] b;",
+                    "qubit[3] q;",
+                    "h q[0];",
+                    "cnot q[0], q[1];",
+                    "cnot q[1], q[2];",
+                    "b[0] = measure q[0];",
+                    "b[1] = measure q[1];",
+                    "b[2] = measure q[2];",
+                ]
+            ),
+            [0, 1, 2],
+        ),
+        (
+            "\n".join(
+                [
+                    "bit[1] b;",
+                    "qubit[3] q;",
+                    "h q[0];",
+                    "h q[1];",
+                    "cnot q[1], q[2];",
+                    "b[0] = measure q[{0, 2}];",
+                ]
+            ),
+            [0, 2],
+        ),
+        (
+            "\n".join(
+                [
+                    "bit[1] b;",
+                    "qubit[2] q;",
+                    "h $0;",
+                    "cnot $0, $1;",
+                    "b[0] = measure $0;",
```

Comment:
Generally, a program can only have physical or virtual qubits (though I'm not sure that's enforced), so this test case shouldn't have `qubit[2] q;`